extensions/fairifier-rdf-extension/deri/grefine/rdf/vocab/VocabularyImporter.java

```diff
@@ -23,171 +25,248 @@
+
+    private static final String PROPERTIES_QUERY_P1 = PREFIXES
+            + "SELECT ?resource ?label ?en_label ?description ?en_description ?definition ?en_definition "
+            + "WHERE { " + "?resource rdf:type rdf:Property. "
+            + "OPTIONAL {?resource rdfs:label ?label.} "
+            + "OPTIONAL {?resource rdfs:label ?en_label. FILTER langMatches( lang(?en_label), \"EN\" )  } "
+            + "OPTIONAL {?resource rdfs:comment ?description.} "
+            + "OPTIONAL {?resource rdfs:comment ?en_description. FILTER langMatches( lang(?en_description), \"EN\" )  } "
+            + "OPTIONAL {?resource skos:definition ?definition.} "
+            + "OPTIONAL {?resource skos:definition ?en_definition. FILTER langMatches( lang(?en_definition), \"EN\" )  } "
+            + "FILTER regex(str(?resource), \"^";
+    private static final String PROPERTIES_QUERY_P2 = "\")}";
+
+    private Repository getModel(String url, String name) throws VocabularyImportException {
+        try {
+            final Repository repository =
+                    new SailRepository(new ForwardChainingRDFSInferencer(new MemoryStore()));
+            repository.initialize();
+            try (RepositoryConnection con = repository.getConnection()) {
+                final ValueFactory F = new ValueFactoryImpl();
+                Model model = getModelFromUrl(url);
+                if (model.getNamespaces().contains(new SimpleNamespace(name, url))) {
+                    con.add(model, F.createIRI(model.getNamespace(name).get().getName()));
+                } else {
+                    try {
+                        con.add(model, F.createIRI(model.getNamespace("").get().getName()));
+                    } catch (NoSuchElementException ex) {
+                        // needed in case the base uri and location are equal, and the base uri is
+                        // not in a name space
+                        con.add(model, F.createIRI(url));
+                    }
+                }
+                con.close();
+                return repository;
+            }
+        } catch (Exception e) {
+            throw new VocabularyImportException("Unable to import vocabulary from " + url, e);
+        }
+    }
+
+    private Model getModelFromUrl(String url)
+            throws IOException, MalformedURLException, IOException, ProtocolException {
+        RDFParser parser = getParserFromUrl(url);
+        HttpURLConnection httpConnection = (HttpURLConnection) new URL(url).openConnection();
+        httpConnection.setRequestMethod("GET");
+        httpConnection.setRequestProperty(HttpHeaders.ACCEPT,
+                parser.getRDFFormat().getDefaultMIMEType());
+        return Rio.parse(httpConnection.getInputStream(), "", parser.getRDFFormat());
+    }
+
+    private RDFParser getParserFromUrl(String url) throws IOException, MalformedURLException {
+        StringBuffer response = new StringBuffer();
```

Comment:
This method is new, it returns a parser of the format that is either returned by the content header of the request or by parsing the content itself full parsing is only done if a plain_text header is returned